src/zarr/codecs/sharding.py

```diff
@@ -615,7 +619,7 @@ def _shard_index_size(self, chunks_per_shard: ChunkCoords) -> int:
             16 * product(chunks_per_shard), self._get_index_chunk_spec(chunks_per_shard)
         )
 
-    @lru_cache
+    @lru_cache  # noqa: B019
```

Comment:
Good to know! Fixed in https://github.com/zarr-developers/zarr-python/pull/1670/commits/530e88b6cd63d4d91c44302f0bc0c6a37749bee4